runtime/src/iree/base/internal/cpu.c

```diff
@@ -310,6 +310,74 @@ static void iree_cpu_initialize_from_platform_x86_64(uint64_t* out_fields) {
   out_fields[0] = out0;
 }
 
+#elif defined(IREE_ARCH_RISCV_64)
+#if defined(IREE_PLATFORM_ANDROID) || defined(IREE_PLATFORM_LINUX)
+
+// First 4 characters are "rv32" or "rv64"
+#define IREE_RISCV_64_ISA_SEARCH_START 4
+
+static void iree_cpu_initialize_from_platform_riscv_64(uint64_t* out_fields) {
+  char* line;
+  char isa[1024];
+  char* cpuinfo_path = "/proc/cpuinfo";
+  FILE* fp;
+  size_t len = 0;
+  ssize_t nread;
+
+  fp = fopen(cpuinfo_path, "r");
+  if (!fp) {
+    return;
+  }
+
+  while ((nread = getline(&line, &len, fp)) != -1) {
+    if (sscanf(line, "isa : %s\n", &isa[0])) {
```

Comment:
Use safe standard library functions that won't overrun buffers: `snscanf` instead of `sscanf`.

Reply:
(Or IREE utility functions from `string_view.h` as linked above).